privacyidea/lib/policydecorators.py

```diff
@@ -290,10 +290,10 @@ def auth_user_passthru(wrapped_function, user_object, passw, options=None):
             policy_object.check_for_conflicts(pass_thru, "passthru")
             pass_thru_action = pass_thru[0].get("action").get("passthru")
             policy_name = pass_thru[0].get("name")
+            g.audit_object.add_policy([p.get("name") for p in pass_thru])
```

Comment:
Can't we just set `write_to_audit_log=True` in the `policies()` call above?
Ok, we would then have the `passhru` entry in the audit even if the user has a token...
Also shouldn't there be only one valid `passthru` action? Even if we have several matching policies with different priorities, only the first one will be used.

Reply:
> Ok, we would then have the `passhru` entry in the audit even if the user has a token...

This is why the policy is only added later to the audit.



> Also shouldn't there be only one valid `passthru` action? Even if we have several matching policies with different priorities, only the first one will be used.

We often (always?) do it this way. 
Why: In theory all these policies would lead to this decision. Even if currently only Policy Prio1. The bonus side effect is, that the admin knows, that Pol-Prio1 and Pol-Prio2 lead to this decision.
If we only would add Pol-Prio1 to the audit log, the admin might think to remove Pol-Prio1 with the expectation, that passthru does not work anymore. But, suprise, it will still work, due to Pol-Prio2, which was hidden before.

Bottomline: I would like to keep it this way.